test/test_nn.py

```diff
@@ -1770,6 +1770,24 @@ def test_Conv2d_inconsistent_types_on_GPU_with_cudnn(self):
         # but it should work with the same type
         nn.functional.conv2d(inputs.float(), weights.float(), bias.float())
 
+    @unittest.skipIf(not TEST_CUDA, 'CUDA not available')
+    @unittest.skipIf(not TEST_CUDNN, 'CUDNN not available')
+    def test_Conv2d_deterministic_on_GPU_with_cudnn(self):
+        inputs = Variable(torch.randn(4, 1, 7, 7).float().cuda())
+        weights = Variable(torch.randn(1, 1, 3, 3).double().cuda())
+        bias = Variable(torch.randn(1).double().cuda())
+
+        torch.backends.cudnn.enabled = True
+        torch.backends.cudnn.benchmark = True
+        torch.backends.cudnn.deterministic = True
```

Comment:
you have to implement a context manager that will restore these flags after the test. right now they will be just left like this forever, destroying the environment for other tests. It needs to be a context manager, so that it's executed even in case of an error.

Reply:
sure, that sounds like a good thing to do although I wondered why the context wasn't managed in rest of the tests...

Reply:
it must have slipped through the cracks. We must have accidentally accepted such PRs, but it's breaking the tests

Reply:
agreed! I'll add the context here to this specific test for now and improve the other tests in separate PR

Reply:
I think it would be useful to expose this context manager outside of the tests, so that in conjunction with the new `torch.random.fork_rng` from https://github.com/pytorch/pytorch/pull/2966 the users can now have an easy way to make their code fully-deterministic.

Reply:
That's a really good idea. Let's do this!